### PR TITLE
Add basic monitoring

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/astronomer/ask-astro/issues/39

an example post success and fail service status
Screenshot 2023-11-01 at 11 27 58 AM

### CODE REVIEW

File: airflow/dags/monitor/monitor.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def check_ui_status():
+    endpoint = "https://ask.astronomer.io"
+    response = requests.get(endpoint)
+    if response.status_code != 200:
+        raise response
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def check_weaviate_status():
```
Comment (sunank200) [inline airflow/dags/monitor/monitor.py:70]: what are we trying to check here?

**Review by sunank200 — COMMENTED** [review]
How will the production DAG post the status of every run?

File: airflow/dags/monitor/monitor.py (hunk 1 above)
Comment (pankajastro) [reply airflow/dags/monitor/monitor.py:70]: 1. weaviate class exist if not then the task fail
2. print the number of record in weaviate class

**Review by sunank200 — APPROVED** [review]

**Review by sunank200 — CHANGES_REQUESTED** [review]
@pankajastro we need the following changes in the DAGs:
- On a regular basis we can post the status on slack. 
- We should run the DAGs like very 10 minutes or so and check if anything is down if it is we should post a status on Slack (and maybe email folks).
Something like [this](https://astronomer.slack.com/archives/C063JH8DCUU/p1698698255031439)  incident can happen any time.

**Comment by pankajastro** [comment]
> * On a regular basis we can post the status on slack.
> * We should run the DAGs like very 10 minutes or so and check if anything is down if it is we should post a status on Slack (and maybe email folks).
As per the current state of PR, it would post on Slack and we can add an env variable for schedule intervals for example `*/10 * * * *`

**Comment by sunank200** [comment]
> > * On a regular basis we can post the status on slack.
> > * We should run the DAGs like very 10 minutes or so and check if anything is down if it is we should post a status on Slack (and maybe email folks).
> 
> As per the current state of PR, it would post on Slack and we can add an env variable for schedule intervals for example `*/10 * * * *`

As discussed in all:
- This schedule should be fine:
- We should post on Slack once a day regardless.
- We should only post during the day immediately if anything breaks or fails.

**Comment by pankajastro** [comment]
@sunank200 I have tested this you can check the sample message in the PR description

**Comment by pankajastro** [comment]
@jedcunningham requesting your feedback on this!

File: airflow/dags/monitor/monitor.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            break
+    if count == 0:
+        print(f"Weavaite class {weaviate_class} is empty!")
+    else:
+        print(f"{count} record found in Weavaite class {weaviate_class}")
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def check_firestore_status():
+    with tempfile.NamedTemporaryFile(mode="w+") as tf:
```
Comment (sunank200) [inline airflow/dags/monitor/monitor.py:91]: We should just add what we are monitoring in firestore as doctoring.

File: airflow/dags/monitor/monitor.py (hunk 1 above)
Comment (sunank200) [reply airflow/dags/monitor/monitor.py:72]: We should just add what we are monitoring in weaviate as doctoring.

File: airflow/dags/monitor/monitor.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        SlackWebhookOperator(
+            task_id="slack_alert",
+            slack_webhook_conn_id=slack_webhook_conn,
+            message=service_status,
+        ).execute(context=context)
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def check_ui_status():
+    endpoint = "https://ask.astronomer.io"
```
Comment (sunank200) [inline airflow/dags/monitor/monitor.py:65]: We should just add what we are monitoring in UI as doctoring.

**Review by sunank200 — COMMENTED** [review]
Overall LGTM. @pankajastro added few comments

**Review by sunank200 — APPROVED** [review]

**Pull request merged**